thunder/tests/opinfos.py

```diff
@@ -2399,6 +2400,35 @@ def pow_sample_input_generator(op, device, dtype, requires_grad, *, no_rhs_numbe
 elementwise_binary_ops.append(zeta_opinfo)
 
 
+def div_sample_generator(op, device, dtype, requires_grad, **kwargs):
+    make = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad, exclude_zero=True)
+    number = partial(make_number, dtype=dtype, exclude_zero=True)
+    shapes = [((4, 2, 3), (4, 2, 3)), ((4, 2, 3), (2, 3)), ((4, 2, 3), (2, 1))]
+    for shape_a, shape_b in shapes:
+        for rounding_mode in (None, "trunc", "floor"):
+            # numerator, denominator, rounding_mode
+            yield SampleInput(make(shape_a), make(shape_b), rounding_mode=rounding_mode)
```

Comment:
There are some dtype mismatches in the test:

```
FAILED thunder/tests/test_ops.py::test_core_vs_torch_consistency_div_nvfuser_cuda_int32 - AssertionError: The values for attribute 'dtype' do not match: torch.float32 != torch.int32.
FAILED thunder/tests/test_ops.py::test_core_vs_torch_consistency_div_nvfuser_cuda_int64 - AssertionError: The values for attribute 'dtype' do not match: torch.float32 != torch.int64.
FAILED thunder/tests/test_ops.py::test_core_vs_torch_consistency_div_nvfuser_cuda_float32 - AssertionError: The values for attribute 'dtype' do not match: torch.float64 != torch.float32.
```

Which is really interesting, and may indicate a bug in some existing type promotion logic. However, we should identify the cases that are causing these errors in this PR, and then consider whether this PR should address them or we should file separate issues for later follow-up.

Do we know if a particular rounding mode, or the tensor x tensor, or the tensor x number, case is the source of these dtype mismatches?

Reply:
Hmm can I ask you how you ran the test case? I am not seeing such error messages. It'll be great if you can help me to replicate this message. Thank you for the feedback!

Reply:
If you have the test requirements installed (nvfuser in particular, but see the [requirements/test.txt](https://github.com/Lightning-AI/lightning-thunder/blob/main/requirements/test.txt)), you can reproduce with
`pytest-3 thunder/tests/test_ops.py::test_core_vs_torch_consistency_div_nvfuser_cuda_float32` .

The fact that it is only nvfuser (and not torch) tests that fail makes me wonder whether it might be in the nvfuser executor or nvfuser itself...

Reply:
Thanks for the information @t-vi ! However, correct me if I am wrong, it doesn't seem that this PR fails the CI testing and such. Shouldn't the error mentioned above fail some tests? Would appreciate your help!

Reply:
I can't see anything immediately incorrect. fyi @kevinstephano 

I think we should skip `test_core_vs_torch_consistency` for nvfuser for now, and file a follow-up issue to investigate these failures:

```
FAILED thunder/tests/test_ops.py::test_core_vs_torch_consistency_div_nvfuser_cuda_int32 - AssertionError: The values for attribute 'dtype' do not match: torch.float32 != torch.int32.
FAILED thunder/tests/test_ops.py::test_core_vs_torch_consistency_div_nvfuser_cuda_int64 - AssertionError: The values for attribute 'dtype' do not match: torch.float32 != torch.int64.
FAILED thunder/tests/test_ops.py::test_core_vs_torch_consistency_div_nvfuser_cuda_float32 - AssertionError: The values for attribute 'dtype' do not match: torch.float64 != torch.float32.
```

Because true division and floor division and trunc are already tested, my guess is that there's an issue with nvFuser's handling of `prims.div`, which I don't think we test separately. A PR adding an OpInfo for `prims.div` could help identify the issue. 

Skipping these tests would let us merge this PR, which has good work, and prioritize fixing the dtype inconsistencies separately.

What do you think, @k223kim ?

Reply:
sounds like there's some promotion logic issue. It's reasonable for us to disable the nvfuser test for now and not block this PR.

But we should open an issue for that and we can follow up with fixing that on the side.

Reply:
Yeah, I can definitely submit a separate issue regarding the nvfuser test!